packages/jupyter-ai-magics/jupyter_ai_magics/magics.py

```diff
@@ -421,7 +421,11 @@ def _append_exchange_openai(self, prompt: str, output: str):
 
     def _decompose_model_id(self, model_id: str):
         """Breaks down a model ID into a two-tuple (provider_id, local_model_id). Returns (None, None) if indeterminate."""
-        if model_id in self.custom_model_registry:
+        # custom_model_registry maps keys to either a model name (a string) or an LLMChain.
+        # If this is an alias to another model, expand the full name of the model.
+        if model_id in self.custom_model_registry and isinstance(
+            self.custom_model_registry[model_id], str
+        ):
             model_id = self.custom_model_registry[model_id]
```

Comment:
Quick comment: it seems like it would be easier to re-define `self.custom_model_registry` to have values also be `LLMChain`s in the case of an alias key. No need to do that in this PR if it's too big of a change though.